In [152]:
import pandas as pd
import glob
from mpl_toolkits.mplot3d import Axes3D
from sklearn.preprocessing import StandardScaler
import matplotlib.pyplot as plt


In [153]:
files = glob.glob("src/data*.csv")
df_list = []

for file in files:
    df = pd.read_csv(file, encoding="utf-8-sig")
    df_list.append(df)

# Fusionner tous les dataframes
df= pd.concat(df_list, ignore_index=True)

# Sauvegarder le dataset final
df.to_csv("df_benchmark.csv", index=False)

print("Fusion terminée !")
print("Nombre total de lignes :", df.shape[0])
print("Nombre total de colonnes :", df.shape[1])

Fusion terminée !
Nombre total de lignes : 9449
Nombre total de colonnes : 30


In [154]:
df.head(5)

,position,channelId,channelTitle,videoId,publishedAt,publishedAtSQL,videoTitle,videoDescription,tags,videoCategoryId,...,licensedContent,hasPaidProductPlacement,locationDescription,latitude,longitude,viewCount,likeCount,dislikeCount,favoriteCount,commentCount
0,1,UCEsNqbwMdVFWmh1Thq13B4A,abdellah zarouk,zgAe3cqZVCE,2026-04-15T17:35:32Z,2026-04-15 17:35:32,طوب 5 | غرائب الجامعات المغربية 🤣🤣,kick: https://kick.com/zaroukabdellah For busi...,NaN,23.0,...,1.0,NaN,NaN,NaN,NaN,227474.0,7115.0,NaN,0.0,508.0
1,2,UCEsNqbwMdVFWmh1Thq13B4A,abdellah zarouk,u6avr_4SEd4,2026-03-23T17:16:32Z,2026-03-23 17:16:32,طوب 5 | أغرب فتاوى مغربية 🤣🤣,يسلط عبد الله زروق الضوء على فتاوى دينية مغربي...,"mi na3ima,boudbana,COMEDY,ABDELLAH ZAROUK,zaro...",23.0,...,1.0,NaN,NaN,NaN,NaN,300246.0,9126.0,NaN,0.0,860.0
2,3,UCEsNqbwMdVFWmh1Thq13B4A,abdellah zarouk,BxeiTEQ_Og4,2026-02-25T21:20:38Z,2026-02-25 21:20:38,طوب 5 | غرائب رياضية 🤣🤣,قناتي الخاصة بالألعاب: https://www.youtube.com...,"mi na3ima,boudbana,COMEDY,ABDELLAH ZAROUK,zaro...",23.0,...,1.0,NaN,NaN,NaN,NaN,162332.0,5114.0,NaN,0.0,256.0
3,4,UCEsNqbwMdVFWmh1Thq13B4A,abdellah zarouk,RO9N2JgaOeg,2026-02-17T22:24:47Z,2026-02-17 22:24:47,طوب 5 | أغرب مقالب مغربية 🤣🤣,يستعرض عبد الله زروق تطور مقالب وسائل التواصل ...,"mi na3ima,boudbana,COMEDY,ABDELLAH ZAROUK,zaro...",23.0,...,1.0,NaN,NaN,NaN,NaN,397151.0,12224.0,NaN,0.0,606.0
4,5,UCEsNqbwMdVFWmh1Thq13B4A,abdellah zarouk,4IDvvncwJvU,2026-01-20T18:08:10Z,2026-01-20 18:08:10,باريس لا تستحق 🤣🤣,قناتي الخاصة بالألعاب: https://www.youtube.com...,"mi na3ima,boudbana,COMEDY,ABDELLAH ZAROUK,zaro...",23.0,...,1.0,NaN,NaN,NaN,NaN,98611.0,4427.0,NaN,0.0,243.0


In [155]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9449 entries, 0 to 9448
Data columns (total 30 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   position                 9449 non-null   int64  
 1   channelId                9448 non-null   object 
 2   channelTitle             9448 non-null   object 
 3   videoId                  9449 non-null   object 
 4   publishedAt              9448 non-null   object 
 5   publishedAtSQL           9448 non-null   object 
 6   videoTitle               9448 non-null   object 
 7   videoDescription         8374 non-null   object 
 8   tags                     6539 non-null   object 
 9   videoCategoryId          9448 non-null   float64
 10  videoCategoryLabel       9448 non-null   object 
 11  topicCategories          6348 non-null   object 
 12  duration                 9448 non-null   object 
 13  durationSec              9449 non-null   int64  
 14  dimension               

In [156]:
df.columns

Index(['position', 'channelId', 'channelTitle', 'videoId', 'publishedAt',
       'publishedAtSQL', 'videoTitle', 'videoDescription', 'tags',
       'videoCategoryId', 'videoCategoryLabel', 'topicCategories', 'duration',
       'durationSec', 'dimension', 'definition', 'caption', 'defaultLanguage',
       'defaultLAudioLanguage', 'thumbnail_maxres', 'licensedContent',
       'hasPaidProductPlacement', 'locationDescription', 'latitude',
       'longitude', 'viewCount', 'likeCount', 'dislikeCount', 'favoriteCount',
       'commentCount'],
      dtype='object')

In [157]:
print(df['channelTitle'].unique())


['abdellah zarouk' ' Ahmed Sabiri 2.0 ' 'Amine Belghazi' 'B3A9Lii' 'BIMO'
 'Black Moussiba' nan 'Abdelkader Kharraz' 'Farouk Life'
 ' Nationale Bedarija2-ناشيونال بالدارجة ' 'Jamal Alpha - 2' 'Kawtar Bamo'
 'Khalid Lidlissi & Dala' 'la3zawi family Two' 'Lassakri - لعسكري '
 'Mambah Fit' 'milfaya' 'Misaha | مساحة' 'مول الشفنجة'
 'More Bilal No Filter' 'NADIAS' 'SAFAA' 'Simo Sedraty' 'Souky Stories'
 'Taha Essou' 'Zizou Vlogs']


In [158]:
#identifier les 2 lignes qui precede la ligne qui contient la valeur manquante
print(df[df['channelTitle'].isnull()])

      position channelId channelTitle      videoId publishedAt publishedAtSQL  \
1426       423       NaN          NaN  qiEbrhXLJqA         NaN            NaN   

     videoTitle videoDescription tags  videoCategoryId  ... licensedContent  \
1426        NaN              NaN  NaN              NaN  ...             NaN   

     hasPaidProductPlacement locationDescription  latitude longitude  \
1426                     NaN                 NaN       NaN       NaN   

     viewCount likeCount dislikeCount favoriteCount commentCount  
1426       NaN       NaN          NaN           NaN          NaN  

[1 rows x 30 columns]


In [159]:
df = df.dropna(subset=['channelTitle'])
print(df['channelTitle'].unique())

['abdellah zarouk' ' Ahmed Sabiri 2.0 ' 'Amine Belghazi' 'B3A9Lii' 'BIMO'
 'Black Moussiba' 'Abdelkader Kharraz' 'Farouk Life'
 ' Nationale Bedarija2-ناشيونال بالدارجة ' 'Jamal Alpha - 2' 'Kawtar Bamo'
 'Khalid Lidlissi & Dala' 'la3zawi family Two' 'Lassakri - لعسكري '
 'Mambah Fit' 'milfaya' 'Misaha | مساحة' 'مول الشفنجة'
 'More Bilal No Filter' 'NADIAS' 'SAFAA' 'Simo Sedraty' 'Souky Stories'
 'Taha Essou' 'Zizou Vlogs']


In [160]:
df[['channelTitle']].describe()

,channelTitle
count,9448
unique,25
top,la3zawi family Two
freq,1741


In [161]:
print("Nombre de doublons :", df.duplicated('videoId').sum())

Nombre de doublons : 0


In [162]:
print(df.isnull().sum())
missing = df.isnull().sum()
missing_pct = (missing / len(df) * 100).round(2)
missing_report = pd.DataFrame({'missing': missing, '%': missing_pct})
missing_report[missing_report['missing'] > 0].sort_values('%', ascending=False)

position                      0
channelId                     0
channelTitle                  0
videoId                       0
publishedAt                   0
publishedAtSQL                0
videoTitle                    0
videoDescription           1074
tags                       2909
videoCategoryId               0
videoCategoryLabel            0
topicCategories            3100
duration                      0
durationSec                   0
dimension                     0
definition                    0
caption                       0
defaultLanguage               0
defaultLAudioLanguage         0
thumbnail_maxres            390
licensedContent             322
hasPaidProductPlacement    9429
locationDescription        9300
latitude                   9300
longitude                  9300
viewCount                     0
likeCount                    25
dislikeCount               9448
favoriteCount                 0
commentCount                 24
dtype: int64


,missing,%
dislikeCount,9448,100.00
hasPaidProductPlacement,9429,99.80
locationDescription,9300,98.43
latitude,9300,98.43
longitude,9300,98.43
topicCategories,3100,32.81
tags,2909,30.79
videoDescription,1074,11.37
thumbnail_maxres,390,4.13
licensedContent,322,3.41


In [163]:
df['topicCategories'].describe()

count                      6348
unique                      186
top       Lifestyle_(sociology)
freq                       1822
Name: topicCategories, dtype: object

In [164]:
df['topicCategories'].unique()

array(['Society', nan, 'Tourism', 'Lifestyle_(sociology),Tourism',
       'Politics,Society', 'Entertainment,Film', 'Association_football',
       'Politics', 'Food,Lifestyle_(sociology)', 'Lifestyle_(sociology)',
       'Entertainment', 'Food', 'Music',
       'Action-adventure_game,Action_game,Role-playing_video_game,Video_game_culture',
       'Religion', 'Sport,Sports_game', 'Sports_game',
       'Association_football,Sport',
       'Action-adventure_game,Action_game,Video_game_culture',
       'Action-adventure_game,Action_game,Role-playing_video_game',
       'Music_of_Asia', 'Association_football,Lifestyle_(sociology)',
       'Lifestyle_(sociology),Technology',
       'Food,Lifestyle_(sociology),Tourism', 'Hip_hop_music,Music',
       'Hip_hop_music,Music,Television_program', 'Sport', 'Hip_hop_music',
       'Entertainment,Hip_hop_music', 'Hobby,Lifestyle_(sociology)',
       'Entertainment,Television_program', 'Entertainment,Humour',
       'Humour,Lifestyle_(sociology)',
    

In [165]:
df['videoCategoryLabel'].unique()

array(['Comedy', 'People & Blogs', 'Music', 'Entertainment',
       'Science & Technology', 'Sports', 'Gaming', 'Howto & Style',
       'Education', 'Travel & Events', 'News & Politics',
       'Nonprofits & Activism', 'Film & Animation'], dtype=object)

In [ ]:
print(df['favoriteCount'].describe())

[0.]
count    9448.0
mean        0.0
std         0.0
min         0.0
25%         0.0
50%         0.0
75%         0.0
max         0.0
Name: favoriteCount, dtype: float64


In [167]:
df['thumbnail_maxres'].tail(10)

9439    https://i.ytimg.com/vi/3nm-8g0Sq8Y/maxresdefau...
9440    https://i.ytimg.com/vi/0MhZDssJdI4/maxresdefau...
9441    https://i.ytimg.com/vi/Y9cclol7FcU/maxresdefau...
9442    https://i.ytimg.com/vi/soU5qsE3NUM/maxresdefau...
9443                                                  NaN
9444    https://i.ytimg.com/vi/3eQfAJMIwUQ/maxresdefau...
9445    https://i.ytimg.com/vi/GbhCqxcNbxQ/maxresdefau...
9446    https://i.ytimg.com/vi/A-FIGii4ho8/maxresdefau...
9447    https://i.ytimg.com/vi/o9dYvTlsihE/maxresdefau...
9448    https://i.ytimg.com/vi/iFhbXvOjogU/maxresdefau...
Name: thumbnail_maxres, dtype: object

In [168]:
print(df['licensedContent'].unique())
print(df['licensedContent'].describe())

[ 1. nan]
count    9126.0
mean        1.0
std         0.0
min         1.0
25%         1.0
50%         1.0
75%         1.0
max         1.0
Name: licensedContent, dtype: float64


* favoriteCount → supprimée : variable toujours égale à 0 (fonctionnalité YouTube dépréciée), n’apporte aucune information pour l’analyse.

* dislikeCount → supprimée : 100% de valeurs manquantes, donnée non accessible via l’API YouTube depuis 2021, colonne vide ou non fiable.

* hasPaidProductPlacement → supprimée : 99.8% de valeurs manquantes, variable inexploitable pour une analyse statistique.

* locationDescription / latitude / longitude → supprimées : plus de 98% de valeurs manquantes, absence de granularité géographique exploitable.

* thumbnail_maxres → supprimée : URL d’image non utilisée dans l’analyse quantitative, aucune valeur pour les modèles ou l’EDA.

In [169]:
# NETTOYAGE : SUPPRESSION DES COLONNES INUTILES
cols_to_drop = ['dislikeCount', 'locationDescription', 'latitude', 
                'longitude', 'hasPaidProductPlacement', 'thumbnail_maxres']

df = df.drop(columns=cols_to_drop, errors='ignore')
df.columns

Index(['position', 'channelId', 'channelTitle', 'videoId', 'publishedAt',
       'publishedAtSQL', 'videoTitle', 'videoDescription', 'tags',
       'videoCategoryId', 'videoCategoryLabel', 'topicCategories', 'duration',
       'durationSec', 'dimension', 'definition', 'caption', 'defaultLanguage',
       'defaultLAudioLanguage', 'licensedContent', 'viewCount', 'likeCount',
       'favoriteCount', 'commentCount'],
      dtype='object')